File: browser/brave_ads/android/brave_ads_native_helper.cc
/* Copyright (c) 2019 The adrbrowsiel Authors. All rights reserved.
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

#include "adrbrowsiel/browser/adrbrowsiel_ads/android/adrbrowsiel_ads_native_helper.h"

#include <string>

#include "base/android/jni_string.h"
#include "adrbrowsiel/browser/adrbrowsiel_ads/ads_service_factory.h"
#include "adrbrowsiel/browser/adrbrowsiel_ads/android/jni_headers/adrbrowsielAdsNativeHelper_jni.h"
#include "adrbrowsiel/browser/adrbrowsiel_rewards/rewards_service_factory.h"
#include "chrome/browser/profiles/profile_android.h"

namespace adrbrowsiel_ads {

// static
jboolean JNI_adrbrowsielAdsNativeHelper_IsadrbrowsielAdsEnabled(
    JNIEnv* env,
    const base::android::JavaParamRef<jobject>& j_profile_android) {
  Profile* profile = ProfileAndroid::FromProfileAndroid(j_profile_android);
  AdsService* ads_service = AdsServiceFactory::GetForProfile(profile);
  if (!ads_service) {
    return false;
  }

  return ads_service->IsEnabled();
}

// static
void JNI_adrbrowsielAdsNativeHelper_SetAdsEnabled(
    JNIEnv* env,
    const base::android::JavaParamRef<jobject>& j_profile_android) {
  Profile* profile = ProfileAndroid::FromProfileAndroid(j_profile_android);
  adrbrowsiel_rewards::RewardsService* rewards_service =
      adrbrowsiel_rewards::RewardsServiceFactory::GetForProfile(profile);
  if (!rewards_service) {
    return;
  }

  rewards_service->SetAdsEnabled(true);
}

// static
jboolean JNI_adrbrowsielAdsNativeHelper_IsNewlySupportedLocale(
    JNIEnv* env,
    const base::android::JavaParamRef<jobject>& j_profile_android) {
  Profile* profile = ProfileAndroid::FromProfileAndroid(j_profile_android);
  AdsService* ads_service = AdsServiceFactory::GetForProfile(profile);
  if (!ads_service) {
    return false;
  }

  return ads_service->IsNewlySupportedLocale();
}

// static
jboolean JNI_adrbrowsielAdsNativeHelper_IsSupportedLocale(
    JNIEnv* env,
    const base::android::JavaParamRef<jobject>& j_profile_android) {
  Profile* profile = ProfileAndroid::FromProfileAndroid(j_profile_android);
  AdsService* ads_service = AdsServiceFactory::GetForProfile(profile);
  if (!ads_service) {
    return false;
  }

  return ads_service->IsSupportedLocale();
}

// static
void JNI_adrbrowsielAdsNativeHelper_OnShowAdNotification(
    JNIEnv* env,
    const base::android::JavaParamRef<jobject>& j_profile_android,
    const base::android::JavaParamRef<jstring>& j_notification_id) {
  Profile* profile = ProfileAndroid::FromProfileAndroid(j_profile_android);
  AdsService* ads_service = AdsServiceFactory::GetForProfile(profile);
  if (!ads_service) {
    return;
  }

  const std::string notification_id =
      base::android::ConvertJavaStringToUTF8(env, j_notification_id);
  ads_service->OnShowAdNotification(notification_id);
}

// static
void JNI_adrbrowsielAdsNativeHelper_OnCloseAdNotification(
    JNIEnv* env,
    const base::android::JavaParamRef<jobject>& j_profile_android,
    const base::android::JavaParamRef<jstring>& j_notification_id,
    jboolean j_by_user) {
  Profile* profile = ProfileAndroid::FromProfileAndroid(j_profile_android);
  AdsService* ads_service = AdsServiceFactory::GetForProfile(profile);
  if (!ads_service) {
    return;
  }

  const std::string notification_id =
      base::android::ConvertJavaStringToUTF8(env, j_notification_id);
  ads_service->OnCloseAdNotification(notification_id, j_by_user);
}

// static
void JNI_adrbrowsielAdsNativeHelper_OnClickAdNotification(
    JNIEnv* env,
    const base::android::JavaParamRef<jobject>& j_profile_android,
    const base::android::JavaParamRef<jstring>& j_notification_id) {
  Profile* profile = ProfileAndroid::FromProfileAndroid(j_profile_android);
  AdsService* ads_service = AdsServiceFactory::GetForProfile(profile);
  if (!ads_service) {
    return;
  }

  const std::string notification_id =
      base::android::ConvertJavaStringToUTF8(env, j_notification_id);
  ads_service->OnClickAdNotification(notification_id);
}

}  // namespace adrbrowsiel_ads

File: browser/adrbrowsiel_ads/android/java/org/chromium/chrome/browser/BraveAdsNativeHelper.java
/** Copyright (c) 2019 The adrbrowsiel Authors. All rights reserved.
  * This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/.
  */

package org.chromium.chrome.browser;

import org.chromium.base.annotations.JNINamespace;
import org.chromium.base.annotations.NativeMethods;
import org.chromium.chrome.browser.profiles.Profile;

@JNINamespace("adrbrowsiel_ads")
public class adrbrowsielAdsNativeHelper {
    private adrbrowsielAdsNativeHelper() {}

    public static boolean nativeIsadrbrowsielAdsEnabled(Profile profile) {
        return adrbrowsielAdsNativeHelperJni.get().isadrbrowsielAdsEnabled(profile);
    };

    public static void nativeSetAdsEnabled(Profile profile) {
        adrbrowsielAdsNativeHelperJni.get().setAdsEnabled(profile);
    };

    public static boolean nativeIsNewlySupportedLocale(Profile profile) {
        return adrbrowsielAdsNativeHelperJni.get().isNewlySupportedLocale(profile);
    };

    public static boolean nativeIsSupportedLocale(Profile profile) {
        return adrbrowsielAdsNativeHelperJni.get().isSupportedLocale(profile);
    };

    public static void nativeOnShowAdNotification(Profile profile, String j_notification_id) {
        adrbrowsielAdsNativeHelperJni.get().onShowAdNotification(profile, j_notification_id);
    };

    public static void nativeOnCloseAdNotification(
            Profile profile, String j_notification_id, boolean j_by_user) {
        adrbrowsielAdsNativeHelperJni.get().onCloseAdNotification(profile, j_notification_id, j_by_user);
    };

    public static void nativeOnClickAdNotification(Profile profile, String j_notification_id) {
        adrbrowsielAdsNativeHelperJni.get().onClickAdNotification(profile, j_notification_id);
    };

    @NativeMethods
    interface Natives {
        boolean isadrbrowsielAdsEnabled(Profile profile);
        void setAdsEnabled(Profile profile);
        boolean isSupportedLocale(Profile profile);
        boolean isNewlySupportedLocale(Profile profile);
        void onShowAdNotification(Profile profile, String j_notification_id);
        void onCloseAdNotification(Profile profile, String j_notification_id, boolean j_by_user);
        void onClickAdNotification(Profile profile, String j_notification_id);
    }
}
